Source/FlowEditor/Private/Graph/Nodes/FlowGraphNode_NamedRerouteDeclaration.cpp

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright https://github.com/MothCocoon/FlowGraph/graphs/contributors
+
+#include "Graph/Nodes/FlowGraphNode_NamedRerouteDeclaration.h"
+#include "FlowAsset.h"
+#include "Graph/FlowGraph.h"
+#include "Graph/Widgets/SFlowGraphNode.h"
+#include "Nodes/Route/FlowNode_NamedRerouteDeclaration.h"
+
+#include UE_INLINE_GENERATED_CPP_BY_NAME(FlowGraphNode_NamedRerouteDeclaration)
+
+UFlowGraphNode_NamedRerouteDeclaration::UFlowGraphNode_NamedRerouteDeclaration(const FObjectInitializer& ObjectInitializer)
+	: Super(ObjectInitializer)
+{
+	AssignedNodeClasses = {UFlowNode_NamedRerouteDeclaration::StaticClass()};
+}
+
+void UFlowGraphNode_NamedRerouteDeclaration::PostEditImport()
+{
+	Super::PostEditImport();
+
+	if (UFlowNode_NamedRerouteDeclaration* Declaration = Cast<UFlowNode_NamedRerouteDeclaration>(GetFlowNodeBase()))
+	{
+		EnsureUniqueNodeTitle();
+		Declaration->CustomNodeColor = FLinearColor::MakeRandomColor();
```

Comment:
This is not necessary, every Declaration node will have its unique color set in constructor

Reply:
When copying or duplicating, the constructor is not called. This is because the copy or duplicate operation does not create a new instance of the node from scratch, but simply duplicates the configuration and state of the existing node. Because of this it is necessary to assign a new color. Please correct me if I am wrong.